omod/src/main/java/org/openmrs/module/patientqueueing/fragment/controller/CompletePatientQueueFragmentController.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.patientqueueing.fragment.controller;
+
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.patientqueueing.PatientQueueingConfig;
+import org.openmrs.module.patientqueueing.api.PatientQueueingService;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.openmrs.module.patientqueueing.utils.QueueingUtil;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import org.openmrs.ui.framework.session.Session;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * VisitIncludes fragment. Retrieve and set the visit type and attributes
+ */
+public class CompletePatientQueueFragmentController {
+	
+	public void controller(@RequestParam("patientId") Patient patient, FragmentModel model, UiSessionContext sessionContext)
+	        throws ParseException {
+		PatientQueue patientQueue = getPreviousQueue(patient, sessionContext.getSessionLocation());
+		Integer currentPatientQueueId = null;
+		if (patientQueue != null) {
+			currentPatientQueueId = patientQueue.getPatientQueueId();
+		}
+		if (currentPatientQueueId != null) {
+			model.put("currentPatientQueueId", currentPatientQueueId);
+		} else {
+			model.put("currentPatientQueueId", "");
+		}
+	}
+	
+	public String post(FragmentModel model,
+	        @RequestParam(value = "patientQueueId", required = false) Integer patientQueueId, Session session, UiUtils ui)
```

Comment:
Why would one post to complete with a null or non existing patientQueueId? In other words, what is the use case for not requiring this param?